libs/ui/src/theme/themeContext.tsx

```diff
@@ -0,0 +1,97 @@
+import { ThemeProvider as ThemeProviderBase } from '@mui/material';
+import React, { PropsWithChildren, useMemo, useState } from 'react';
+import getTokensByMode from './helpers/getTokensByMode';
+import isDarkMode from './helpers/isDarkMode';
+import useSynchronizeThemeAndMedia from './hooks/useSynchronizeThemeAndMedia';
+import getMuiCustomTheme from './helpers/getMuiCustomTheme';
+import Theme, { PartialTheme } from './themeContext.types';
+import { CssBaseline } from '@mui/material';
+
+const defaultLightValue = getTokensByMode('light');
+const defaultDarkValue = getTokensByMode('dark');
+
+const themeContext = React.createContext<Theme>(defaultLightValue);
+
+export type ThemeProviderProps = PropsWithChildren & {
+  theme?: {
+    lightMode: PartialTheme;
+    darkMode?: PartialTheme;
+  };
+};
+
+export const ThemeProvider: React.FC<PropsWithChildren<ThemeProviderProps>> = ({
+  children,
+  theme,
+}) => {
+  const themeSource: {
+    light: Theme;
+    dark: Theme;
+  } = useMemo(() => {
+    return {
+      light: mergeThemeConfigurations({
+        defaultValue: defaultLightValue,
+        overrides: theme?.lightMode ?? {},
+      }),
+      dark: mergeThemeConfigurations({
+        defaultValue: defaultDarkValue,
+        overrides: theme?.darkMode ?? {},
+      }),
+    };
+  }, [theme]);
+
+  const [tokens, setTokens] = useState<Theme>(() => {
+    return isDarkMode() ? themeSource.dark : themeSource.light;
+  });
+
+  const muiTheme = getMuiCustomTheme({ tokens });
```

Comment:
Let wrap this in **useMemo** so the object is only recomputed when **tokens** change:

```suggestion
  const muiTheme = useMemo(() => getMuiCustomTheme({ tokens }), [tokens]);
```